### PR TITLE
Upgrade to Ruby 3

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Hell yeah, it's time for **Ruby 3** and the build is green. Hot stuff. :hot_pepper::hot_pepper::hot_pepper:

Soundtrack: [Quando los pies besan el piso](https://www.youtube.com/watch?v=F0naf0cThd4) :musical_note: 

#### What should we test?

- Deploying and doing a quick smoke test to see if the app is running. 
- Check downloading reports as spreadsheets (CSV) is working.

Note: requires provisioning with https://github.com/openfoodfoundation/ofn-install/pull/791 before deployment. UK Staging now has Ruby 3.0.3 installed, so this PR can be deployed there.

#### Release notes

Changelog Category: Technical changes

### CODE REVIEW

File: Gemfile
Comment (jibees) [inline Gemfile:97]: 🤓

**Review by jibees — APPROVED** [review]
toy-story-buzz-lightyear

File: Gemfile
Comment (mkllnk) [inline Gemfile:4]: What do you think about this?

```suggestion
ruby File.read(".ruby-version").chomp
```

File: Gemfile
Comment (mkllnk) [reply Gemfile:97]: Hm, makes me wonder if we are a bit early to the party. This is additional work. Hopefully it pays off.

File: Gemfile.lock
Comment (mkllnk) [inline Gemfile.lock:172]: Just because? Or for compatibility?

**Review by mkllnk — APPROVED** [review]
Nice! I hope we'll see some performance improvements.

You can decide about my suggestion below.

File: Gemfile.lock
Comment (Matt-Yorkley) [reply Gemfile.lock:172]: Compatibility. The new Bootsnap version was updated for Ruby 3.

File: Gemfile
Comment (Matt-Yorkley) [reply Gemfile:4]: Is it worth doing `#strip` instead?

File: Gemfile
Comment (Matt-Yorkley) [reply Gemfile:97]: Ruby 3 has been around for over a year. The maintainers of the `roo` gem have just been a bit slow on updating/releasing.

File: Gemfile
Comment (Matt-Yorkley) [reply Gemfile]: Updated :+1:

Comment (mkllnk) [reply Gemfile]: There's not leading whitespace in the file. So `chomp` seems to be more accurate for this case but it doesn't really matter. :shrug:

**Review by mkllnk — APPROVED** [review]

**Comment by filipefurtad0** [comment]
Hey @Matt-Yorkley,
A quick ping on this one: I've tried to deploy it on staging-UK, but getting:
https://semaphoreci.com/openfoodfoundation/openfoodnetwork-2/servers/uk-staging/deploys/1075

**Comment by Matt-Yorkley** [comment]
It deploys now :rocket:

**Comment by filipefurtad0** [comment]
@Matt-Yorkley ,
I think we're good :+1: 
Tested:
- backend, frontend (visually)
- some integrations, like:
  - maps
  - payment method (Stripe and Paypal)
- some translations - `t(')` and `js`
- image upload

I tried to "dodge" the tests already existing automatic tests - which are green - and use real data instead. Found nothing, this looks great! :rocket:

**Comment by filipefurtad0** [comment]
I'm moving to ready to go, and not yet merging - as I understood this would prevent staging in other servers other than staging-UK :+1: Once we've provisioned them we can merge, does this sound good?

**Pull request merged**
